timelock-corruption/src/main/java/com/palantir/history/models/CompletePaxosHistoryForNamespaceAndUsecase.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.history.models;
+
+import java.util.List;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+import org.immutables.value.Value;
+
+import com.palantir.paxos.Client;
+
+@Value.Immutable
+public interface CompletePaxosHistoryForNamespaceAndUsecase {
+    @Value.Parameter
+    Client namespace();
+
+    @Value.Parameter
+    String useCase();
+
+    @Value.Parameter
+    List<ConcurrentSkipListMap<Long, LearnedAndAcceptedValue>> localAndRemoteLearnerAndAcceptorRecords();
```

Comment:
it's not normal to depend on concrete collections, so is there a reason we'd mandate a ConcurrentSkipListMap?